### ISSUE TITLE
HTML Report is completely blank >= Xcode 16 beta 3

### ISSUE THREAD

**Issue by dokimyj** [XCTestHTMLReport/XCTestHTMLReport#358]
Hello, 

I have an issue with `xchtmlreport` execution when testing using Xcode 16 beta 3 & 4.

In detail, `index.html` from `xchtmlreport` command is completely blank even though there exists a `.xcresult` file.
Here is what I get when executing `xchtmlreport` with full normal `.xcresult` file:

```
# Same thing happens when using Xcode 16 beta 4 as DEVLEOPER_PATH
$ DEVELOPER_DIR=/Applications/Xcode_16_beta_3.app ./xchtmlreport fastlane/test_output/myapptest.xcresult
Error: This command is deprecated and will be removed in a future release, --legacy flag is required to use it.
Usage: xcresulttool get object [--legacy] --path <path> [--id <id>] [--version <version>] [--format <format>]
  See 'xcresulttool get object --help' for more information.
Warning: Can't find invocation record for : fastlane/test_output/myapptest.xcresult

Report successfully created at /path/to/source/fastlane/test_output/index.html
```

<img width="1448" alt="スクリーンショット 2024-07-30 10 57 02" src="https://github.com/user-attachments/assets/79df921e-b848-4b2f-839b-17a2031fd418">

The screenshot above is index.html after the command executed.
As you see, it is completely blank.

Works well when using Xcode 16 beta 2 

```
$ DEVELOPER_DIR=/Applications/Xcode_16_beta_2.app ./xchtmlreport fastlane/test_output/myapptest.xcresult
Warning: Can't find test reference for action Build "myapptest"
Warning: Can't find test reference for action Build "myapptest"
Warning: Can't find test reference for action Build "myapptest"
Warning: Can't find test reference for action Build for Testing "myapptest"

Report successfully created at /path/to/source/fastlane/test_output/index.html
```

<img width="268" alt="スクリーンショット 2024-07-30 11 07 27" src="https://github.com/user-attachments/assets/fb8ec046-9caa-4784-ae45-4f805d8c82b5">
(Some warnings shown up but HTML report itself is normal)

I found the fact out that XCTestHTMLReport is using `XCResultKit` derived from the fork of the [original](https://github.com/davidahouse/XCResultKit), 
which has been updated along with Xcode 16 beta releases.

Here is the diff between original `XCResultKit` and the fork, which is a dependency of this repository.

https://github.com/tylervick/XCResultKit/compare/main...davidahouse%3AXCResultKit%3Amain#diff-323dfc344bcc9a79dc779a072e206692565a32efdb3c2c9e2d24c8e80cdd5e8dR126-R142

Hope this one fixed soon. Thanks in advance.

**Comment by juanitos23** [comment]
+1

**Comment by juanitos23** [comment]
Hello! Replacing `XCResultKit` with from https://github.com/tylervick/XCResultKit.git to https://github.com/davidahouse/XCResultKit.git should resolve the issue. This one has support for xcode16 beta3. I can submit a PR, if I can get write access so I can create a branch.

**Comment by dokimyj** [comment]
Confirmed the issue has been resolved in 2.5.0. Thanks a lot!
